File: src/main/java/io/funky/fangs/keep_it_personal/command/FillDeathPreferencesCommand.java
package io.funky.fangs.keep_it_personal.command;

import com.mojang.brigadier.context.CommandContext;
import io.funky.fangs.keep_it_personal.domain.DeathPreferenceContainer;
import net.minecraft.server.command.ServerCommandSource;

import static net.minecraft.text.Text.literal;

public class FillDeathPreferencesCommand extends KeepItPersonalCommand {
    public static final String NAME = "everything";
    public static final String PERMISSION = KeepItPersonalCommand.PERMISSION + '.' + NAME;
    private static final String RESPONSE = "Updated your preferences!";

    @Override
    public Integer run(CommandContext<ServerCommandSource> context, DeathPreferenceContainer container) {
        container.fillDeathPreferences();
        context.getSource().sendFeedback(() -> literal(RESPONSE), false);
        return SINGLE_SUCCESS;
    }
}

File: src/main/kotlin/io/funky/fangs/keep_it_personal/command/FillDeathPreferencesCommand.kt
package io.funky.fangs.keep_it_personal.command

import com.mojang.brigadier.Command
import com.mojang.brigadier.context.CommandContext
import net.minecraft.server.command.ServerCommandSource
import net.minecraft.server.network.ServerPlayerEntity
import net.minecraft.text.Text

class FillDeathPreferencesCommand: KeepItPersonalCommand() {
    companion object {
        const val NAME = "everything"
    }

    override fun run(
        keepCommandState: KeepCommandState,
        context: CommandContext<ServerCommandSource>,
        player: ServerPlayerEntity
    ): Int {
        keepCommandState.fillPlayerPreferences(player.uuid)
        context.source.sendFeedback({ Text.literal("Updated your preferences!") }, false)
        return Command.SINGLE_SUCCESS
    }
}
